hibernate-core/src/main/java/org/hibernate/resource/transaction/backend/jdbc/internal/JdbcIsolationDelegate.java

```diff
@@ -102,4 +103,18 @@ else if ( e instanceof SQLException ) {
 			throw sqlExceptionHelper().convert( sqle, "unable to obtain isolated JDBC connection" );
 		}
 	}
+
+	@Override
+	public <T> T delegateCallable(Callable<T> callable, boolean transacted) throws HibernateException {
```

Comment:
I think we should have some validation of transacted here.  Passing transacted=true makes no sense if we are not getting a new Connection.